v-next/hardhat/src/internal/helpers/config-loading.ts

```diff
@@ -0,0 +1,61 @@
+import { isAbsolute, resolve } from "node:path";
+import { pathToFileURL } from "node:url";
+
+import { HardhatError } from "@nomicfoundation/hardhat-errors";
+import { findUp } from "@nomicfoundation/hardhat-utils/fs";
+
+import { ERRORS } from "../../../../hardhat-errors/src/descriptors.js";
+
+async function findClosestHardhatConfig(): Promise<string> {
+  let hardhatConfigPath = await findUp("hardhat.config.ts");
+
+  if (hardhatConfigPath !== undefined) {
+    return hardhatConfigPath;
+  }
+
+  hardhatConfigPath = await findUp("hardhat.config.js");
+
+  if (hardhatConfigPath !== undefined) {
+    return hardhatConfigPath;
+  }
+
+  throw new HardhatError(ERRORS.GENERAL.NO_CONFIG_FILE_FOUND);
+}
+
+export async function resolveConfigPath(): Promise<string> {
+  const configPath = process.env.HARDHAT_CONFIG;
+
+  if (configPath !== undefined) {
+    return configPath;
+  }
+
+  return findClosestHardhatConfig();
+}
+
+export async function importUserConfig(configPath: string) {
+  const normalizedPath = isAbsolute(configPath)
+    ? configPath
+    : resolve(process.cwd(), configPath);
+
+  const { exists } = await import("@nomicfoundation/hardhat-utils/fs");
+
+  if (!(await exists(normalizedPath))) {
+    throw new HardhatError(ERRORS.GENERAL.INVALID_CONFIG_PATH, { configPath });
+  }
+
+  const imported = await import(pathToFileURL(normalizedPath).href);
+
+  if (!("default" in imported)) {
+    throw new HardhatError(ERRORS.GENERAL.NO_CONFIG_EXPORTED, { configPath });
+  }
+
+  const config = imported.default;
+
+  if (typeof config !== "object" || config === null) {
```

Comment:
@alcuadrado This would allow `Array` configs to be passed, is that intended? By the way, this is the second time I've seen an `isObject` check (which is also missing the `Array` check). I believe we should add this as a util.

Reply:
@zoeyTM you can ignore this, I'll put a separate PR adding `isObject` and replacing the occurrences with the helper.